core/src/Cabana_Sort.hpp

```diff
@@ -589,6 +589,195 @@ void permute( const BinningDataType& binning_data,
     Kokkos::fence();
 }
 
+//---------------------------------------------------------------------------//
+/*!
+  \brief Given binning data and range permute part of an AoSoA.
+
+  \tparam BinningDataType The binning data type.
+
+  \tparm AoSoA_t The AoSoA type.
+
+  \param binning_data The binning data.
+
+  \param aosoa The AoSoA to permute.
+
+  \param begin The beginning index of the AoSoA range to sort.
+
+  \param end The end index of the AoSoA range to sort.
+*/
+template<class BinningDataType, class AoSoA_t>
+void permute( const BinningDataType& binning_data,
+              AoSoA_t& aosoa,
+              const std::size_t begin,
+              const std::size_t end,
+              typename std::enable_if<(is_binning_data<BinningDataType>::value &&
```

Comment:
Per @dalg24 comment before we can replace these with `static_assert` in the function body.